### DISCUSSION TITLE
road warrior setup with charon-systemd or network manager applet ?

### DISCUSSION THREAD

**Discussion by p088gll** [strongswan/strongswan#2182] [General]
We are in a dual-stack IPv4/6 environment and use Ubuntu Laptops for road warrior access via IPSEC and a Fortigate as IPSEC endpoint. Certificate based authentication is via a yubikey secured with a PIN.

Since the strongswan nm applet in Ubuntu 22.04 doesnt support IPv6, we ended up using charon-systemd and vici/swanctl. There are a few problems with this:

In the standard setup charon-systemd sets up a unix socket for vici, which is only accessible by root. Therefore all users need to have sudo permissions to enter their yubikey PIN with swanctl --load-creds and start the tunnels. Also, monitoring the tunnel with swanctl --list-sas requires sudo. We do this with a KDE plasmoid every 2 minutes. 

Making charon-systemd listen on a network socket on localhost allows access without sudo. However, then every local user or application can access the socket, which is less safe to me.  Since every human user in our organisation has sudo permissions anyway, it might still be OK.

I heard that is is also possible to let charon-systemd run as a non-root user. But I am not sure how messy this could get, so I havent followed this path yet. The goal would be that the logged-in user can work with swanctl and start/stop charon-systemd without the need for sudo.

With the network manager applet in Ubuntu 24.04. there should be IPv6 support and configuration of charon is with D-Bus. So there should be no more problems with sudo. However, I am not sure if communication with a fortigate endpoint will work.

My main question/concern is: Should we switch to the network manager applet with Ubuntu 24.04 ? Are there more people around here being happy using charon-systemd for road-warrior setups (so we dont need to switch) ?

**Comment by tobiasbrunner** [comment]
> Since the strongswan nm applet in Ubuntu 22.04 doesnt support IPv6

As discussed in #2179, that's not the case.

> I heard that is is also possible to let charon-systemd run as a non-root user.

You can also just change the group (`charon.group`) without changing the user. Then the UNIX sockets will all be owned by that group and users that belong to it will be able to connect.

**Reply by p088gll** [reply]
charon.group is exactly what we need. Works great. Thank you.
